test/spec/api/v4/dataview/category.spec.js

```diff
@@ -169,7 +160,7 @@ describe('api/v4/dataview/category', function () {
       dataview.on('limitChanged', limitChangedSpy);
 
       expect(limitChangedSpy).not.toHaveBeenCalled();
-      dataview.$setEngine(createEngineMock());
+      dataview.$setEngine(createEngine());
```

Comment:
Why not using MockFactory.createEngine? This is used more than 20 times. IMO we should use only one of these (mockFactory OR engine.fixture)

Reply:
I was aiming the other way round, using the fixture for everything instead of getting a fat MockFactory.

I'll open a separated issue to unify this but I didn't want to clutter the PR even more.

Reply:
Agree :+1: